walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,175 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+
 
 public final class FindMeetingQuery {
+  /**
+   * Handles "find a meeting" feature by using the exsisting API to return the times
+   * when the meeting could happen that day, given the meeting information.
+   *
+   * @param events all events happening on a given day including event name, 
+   *                a TimeRange, and collection of attendees.
+   *
+   * @param request details of meeting being requested including requested meeting name, 
+   *                duration, and collection of attendees.
+   *
+   * @return an ArrayList containing all the TimeRanges the request meeting can be scheduled
+   *
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    long duration = request.getDuration();
+
+    if (duration > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    } else if (events.isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    List<String> attendees = new ArrayList<String>(request.getAttendees());
+    List<TimeRange> busyTimeRanges = findBusyTimes(events, attendees);
+
+    if (!request.getOptionalAttendees().isEmpty()) {
+      List<String> optional = new ArrayList<String>(request.getOptionalAttendees());
+      attendees.addAll(optional);
+
+      List<TimeRange> optionalBusyTimeRanges = findBusyTimes(events, attendees);
+      List<TimeRange> optionalMeetingTimes = findMeetingTimes(optionalBusyTimeRanges, duration);
+
+      if(!optionalMeetingTimes.isEmpty()) {
+        return optionalMeetingTimes;
+      } else {
+        return Arrays.asList();
+      }
+
+    }
+
+    return findMeetingTimes(busyTimeRanges, duration);
+  }
+
+  //finds meeting times that don't fall during busy times of the day
```

Comment:
nit: update comments to be full sentences. Space between // and first letter, capitalize first letter and end with period. 

It doesn't directly say to do this in the style guide, but all examples and other usages from the style guide are like this.
https://google.github.io/styleguide/javaguide.html#s4.8.6-comments